fsserver/fsserver.c

```diff
@@ -291,6 +297,57 @@ void install_shims() {
 	}
 }
 
+char *log_path() {
+	char *path = NULL;
+	char time_str[256];
+
+	time_t ti         = time(NULL);
+	struct tm *tm     = gmtime(&ti);
+	struct passwd *pw = getpwuid(getuid());
+
+	size_t n = strftime(time_str, sizeof(time_str), "%Y.%b.%d.%H.%M.%S", tm);
+	/* the second case is supposedly for very old, <= 4.4.1 libc, and maybe some more until
+	 * 4.4.4 */
+	if (n == 0 || n >= sizeof(time_str))
+		fatal("making fsserver.err file name", "strftime");
+
+	if (asprintf(&path, "%s/fsserver.%s.err", pw->pw_dir, time_str) < 0)
+		fatal("making fsserver.err file format string", strerror(errno));
+
+	return path;
+}
+
+void setup_daemon_log(char *path) {
+	int fd_err = open(path, O_WRONLY | O_CREAT | O_EXCL, S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH);
+	if (fd_err < 0) {
+		fatal("opening fsserver.err file", strerror(errno));
+	}
+
+	if (dup2(fd_err, STDOUT_FILENO) < 0 || dup2(fd_err, STDERR_FILENO) < 0)
+		fatal("connecting to fsserver.err file", strerror(errno));
+}
+
+void setup_foreground_log(char *path) {
+	char *linep = NULL;
+	if (asprintf(&linep, "/usr/bin/tee %s", path) < 0)
+		fatal("making tee command", "asprintf");
```

Comment:
You changed the other instance of this error check to use `strerror(errno)` in the `fatal()` call. Was not changing it here on purpose?

I had not used `strerror(errno)` because the _man_ page for `asprintf()` does not mention setting `errno` on an error whereas typically other _man_ pages do. Perhaps that is just a "typo."

Reply:
Good point. Looking into the code, it does set errno (via the other libc functions it call), so it should give something sensible.